### PR TITLE
Feature/baf_upd

### PR DESCRIPTION
**Pull request by melferink**
* Include refset database
* Included family IGV session
* Updated single sample IGV session

### CODE REVIEW

File: ExomeDepth/exomedepth_db.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(sample_id, flowcell_id, args.refset)
+            store_sample(Session, entry)
+        refset = (
+            session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+        if args.print_refset:
+            print(refset)
+        return refset
```
Comment (rernst) [inline ExomeDepth/exomedepth_db.py:135]: Function is almost identical to `add_sample_to_db_and_return_refset`. Might want to merge overlapping code?

File: ExomeDepth/exomedepth_db.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+
+def determine_sample_id(bam):
```
Comment (rernst) [inline ExomeDepth/exomedepth_db.py:138]: rename to `get_sample_id` like `get_flowcell_id`

File: ExomeDepth/igv_session_template_family.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (rernst) [inline ExomeDepth/igv_session_template_family.xml:1]: Very hard to review, so no comments...

File: ExomeDepth/igv_xml_session.py [hunk 4]
```diff
@@ -1,59 +1,333 @@
...
-                    'bam':bam, 'vcf_hc':vcf_hc, 'sample_id':sample_id, 'igv_ed_hc_test':igv_ed_hc_test, 
-                    'igv_ed_umcu_test':igv_ed_umcu_test, 'bam_coverage':bam_coverage, 'bam_junctions':bam_junctions,  
-                    'vcf_SNV':vcf_SNV, 'min_axis':min_axis, 'mid_axis':mid_axis, 'max_axis':max_axis, 
-                    'ratioid_UMCU':ratioid_UMCU, 'ratioid_HC':ratioid_HC 
-                   }
+
+def parse_ped_family(ped_file, sampleid):
+    sample_dic = {}
+    for line in ped_file:
+        splitline = line.split()
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:12]: No `strip()` needed?

File: ExomeDepth/igv_xml_session.py [hunk 5]
```diff
@@ -1,59 +1,333 @@
...
+        if splitline[1] not in sample_dic:
+            sample_dic[splitline[1]] = [splitline[0], splitline[2], splitline[3]]
+    """ Return familyID, child, father, mother """
+    return sample_dic[sampleid][0], sampleid, sample_dic[sampleid][1], sample_dic[sampleid][2]
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:16]: You are parsing the entire file while you would only like to have the info from `sample_id`

```python
def parse_ped_family(ped_file, sample_id):
 for line in ped_file:
    family, sample, father, mother = line.split()[0:4]
    if sample == sample_id:
        return family, sample_id, father, mother

```

File: ExomeDepth/igv_xml_session.py [hunk 6]
```diff
@@ -1,59 +1,333 @@
...
+        args.refset, args.sampleid, args.runid, igv_extension
+    )  # CNV igv session with ratios for UMCU
+
+    """ ID XML variables"""
+    session = "{0}_{1}_{2}_igv.xml".format(sample_id, statistic, args.runid)  # Session ID
+    hc_cnv_vcf_id = "CNV:HC_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, vcf_extension
+    )  # HC CNV -VCF track id in IGV
+    snv_vcf_id = "SNV/MNV:{0}".format(snv_vcf.split("/")[-1])  # SNV/MNV track ID in IGV
+    igv_hc_ratio_track = "{0}_{1}_test".format(igv_hc_ratio, statistic)  # Ratio track within HC CNV igv session
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:45]: `_test` -> is this correct?

File: ExomeDepth/igv_xml_session.py [hunk 7]
```diff
@@ -1,59 +1,333 @@
...
+        'min_axis': min_axis,
+        'baf_track': baf_track,
+        'baf_track_id': baf_track_id,
+        'igv_umcu_ratio_track': igv_umcu_ratio_track,
+        'igv_umcu_ratio_track_id': igv_umcu_ratio_track_id,
+        'bam_coverage': bam_coverage,
+        'bam_id': bam,
+        'sample_id': sample_id,
+        'fontsize': fontsize
+    }
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:81]: All these variables are created above and not used after that. Why not directly add them to the dict?

```
 substitute_dic = {
    'session': "{0}_{1}_{2}_igv.xml".format(sample_id, statistic, args.runid)
    ....
}

File: ExomeDepth/igv_xml_session.py [hunk 8]
```diff
@@ -1,59 +1,333 @@
...
     new_file = template_file.substitute(substitute_dic)
     return new_file
 
-if __name__ == "__main__":
 
+def make_family_igvsession(args, statistic, igv_extension, vcf_extension, familyid, child, father, mother, child_bam, refsets):
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:86]: `igv_session`

File: ExomeDepth/igv_xml_session.py [hunk 9]
```diff
@@ -1,59 +1,333 @@
...
+    igv_umcu_ratio = "igv_tracks/UMCU_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, igv_extension
+    )  # CNV igv session with ratios for UMCU
+
+    """ ID XML variables"""
+    session = "{0}_{1}_{2}_igv.xml".format(sample_id, statistic, args.runid)  # Session ID
+    hc_cnv_vcf_id = "CNV:HC_{0}_{1}_{2}_{3}".format(
+        args.refset, args.sampleid, args.runid, vcf_extension
+    )  # HC CNV -VCF track id in IGV
+    snv_vcf_id = "SNV/MNV:{0}".format(snv_vcf.split("/")[-1])  # SNV/MNV track ID in IGV
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:44]: You are using this pattern of getting the file name a lot -> maybe create a separate function and reuse.

File: ExomeDepth/igv_xml_session.py [hunk 10]
```diff
@@ -1,59 +1,333 @@
...
+        'baf_track_child': baf_track_child,
+        'baf_track_child_id': baf_track_child_id,
+        'bam_coverage': bam_coverage,
+        'bam_id': child_bam
+    }
+    new_file = template_file.substitute(substitute_dic)
+    return new_file
+
+
+def make_single(args, igv_extension, vcf_extension):
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:216]: make single what?

File: ExomeDepth/igv_xml_session.py [hunk 11]
```diff
@@ -1,59 +1,333 @@
...
+    else:  # Use relative path
+        if args.pipeline == "iap":  # For analysis based on IAP
+            bam = "{0}_dedup.realigned.bam".format(args.sampleid.split(",")[0])  # BAM file
+        elif args.pipeline == "nf":  # For NF pipeline.
+            bam = "{0}.bam".format(args.sampleid.split(",")[0])  # Bam file
+
+    for statistic in settings.igv_settings:
+        write_file = open("{0}/{1}_{2}_{3}_igv.xml".format(
+            args.output, args.sampleid, statistic, args.runid), "w"
+        )
```
Comment (rernst) [inline ExomeDepth/igv_xml_session.py:228]:
```python
write_file = open(
    "{0}/{1}_{2}_{3}_igv.xml".format(args.output, args.sampleid, statistic, args.runid), 
    "w"
)
```
or
```python
write_file = open(
    "{0}/{1}_{2}_{3}_igv.xml".format(
        args.output, args.sampleid, statistic, args.runid
    ), 
    "w"
)
```

File: ExomeDepth/run_ExomeDepth.py [hunk 12]
```diff
@@ -198,132 +204,174 @@ def multiprocess_call(multiprocess_list):
...
+            store_sample(Session, entry)
+            return True
+        else:
+            return False
+
+
+def query_refset(sample_id, flowcell_id):
+    Session = connect_database()
+    with Session() as session:
+        return session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:228]: These function are or should be in `exomedepth_db.py`?

File: ExomeDepth/run_ExomeDepth.py [hunk 13]
```diff
@@ -198,132 +204,174 @@ def multiprocess_call(multiprocess_list):
...
 
     """Call CNV from BAMs"""
     bam = format(os.path.abspath(args.inputbam))
     output_folder = format(os.path.abspath(args.output))
     analysis = settings.analysis
-    prob = settings.probability
+
     if not os.path.isdir(output_folder):
-         os.system("mkdir -p " + str(output_folder)) 
+        os.system("mkdir -p " + str(output_folder))
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:239]: Use https://docs.python.org/3/library/os.html#os.makedirs ?

File: ExomeDepth/run_ExomeDepth.py [hunk 14]
```diff
@@ -198,132 +204,174 @@ def multiprocess_call(multiprocess_list):
...
+
+    multiprocess_list = []
     for model in analysis:
-        multiprocess_list += [[model, output_folder, gender, analysis[model]["target_bed"], analysis[model]["exon_bed"], bam, output_folder, analysis[model]["calling_model"]]] 
-    
+        multiprocess_list += [[
+            model, output_folder, gender, analysis[model]["target_bed"],
+            analysis[model]["exon_bed"], bam, output_folder, refset
+            ]
+        ]
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:264]: Complex nested list? Maybe used a list of dictionaries?

File: ExomeDepth/submit_batch_exomedepth.py [hunk 15]
```diff
@@ -1,41 +1,57 @@
...
+
+    os.system("mkdir -p {output}/{sample}".format(
+        output=args.outputfolder, sample=sampleid)
+    )
+    os.system("ln -sd {bam} {output}/{sample}/{bamfile}".format(
+        bam=bam, bamfile=bamfile, sample=sampleid, output=args.outputfolder)
+    )
+    os.system("ln -sd {bam}.bai {output}/{sample}/{bamfile}.bai".format(
+        bam=bam, bamfile=bamfile, sample=sampleid, output=args.outputfolder)
+    )
```
Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:39]: https://docs.python.org/3/library/os.html#os.symlink

File: ExomeDepth/submit_batch_exomedepth.py [hunk 16]
```diff
@@ -88,59 +148,76 @@ def process(bam, args, refset_dic, gender_dic, suffix_dic):
...
     """Start exomedepth re-analysis"""
     with Pool(processes=int(args.simjobs)) as pool:
-        result = pool.starmap(process, [[bam, args, refset_dic, gender_dic, suffix_dic] for bam in bams])
+        sampleinfo = pool.starmap(exomedepth_analysis, [[bam, args, gender_dic, suffix_dic, refset_dic] for bam in bams])
 
     """ Make CNV summary file """
     logs = glob.glob("{outputfolder}/logs/HC*stats.log".format(outputfolder=args.outputfolder), recursive=True)
     if not os.path.isdir("{inputfolder}/QC/CNV/".format(inputfolder=args.inputfolder)):
         os.system("mkdir -p {inputfolder}/QC/CNV/".format(inputfolder=args.inputfolder))
+        pass
```
Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:220]: Why is this needed?

File: ExomeDepth/submit_batch_exomedepth.py [hunk 17]
```diff
@@ -149,3 +226,20 @@ def process(bam, args, refset_dic, gender_dic, suffix_dic):
...
+        os.system(action)
+
+    """ Make family IGV session(s)"""
+    families = parse_ped(open(args.pedfile, "r"))
+    for item in sampleinfo:
+        if len(list(set(families[item[0]]["parents"]))) == 2:
+            action = "python {0}/igv_xml_session.py family_igv {1} {2} {3} {4} {5}".format(
+                settings.cwd, args.outputfolder, args.pedfile, args.runid, item[0], " ".join(bams)
+            )
+            print(action)
```
Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:244]: Debug code?

**Review by rernst — CHANGES_REQUESTED** [review]
General remarks:
- Create (sub) folders for ExomeDepth, it is getting a bit messy with all the different files.
  - Sugestion: templates, database, utils? 
- Made some (style) comments for print, if and query -> I made these once but can be applied multiple times.

File: ExomeDepth/check_gender_bam.py [hunk 18]
```diff
@@ -28,15 +32,44 @@ def get_gender(bam, arguments):
...
-    parser.add_argument('--gender_determination_y_ratio_female', default=settings.gender_determination_y_ratio[0], type=float, help='Maximum Y ratio threshold females (default = gender_determination_y_ratio[0] in settings.py)')
-    parser.add_argument('--gender_determination_y_ratio_male', default=settings.gender_determination_y_ratio[1], type=float, help='Minimum Y ratio threshold males (default = gender_determination_y_ratio[1] in settings.py)')
-    parser.add_argument('--gender_determination_x_ratio_male', default=settings.gender_determination_x_ratio[0], type=float, help='Maximum X ratio threshold males (default = gender_determination_x_ratio[0] in settings.py)')
-    parser.add_argument('--gender_determination_x_ratio_female', default=settings.gender_determination_x_ratio[1], type=float, help='Minimum X ratio threshold females (default = gender_determination_x_ratio[1] in settings.py)')
+    parser.add_argument(
+        '--gender_determination_locus_y',
+        default=settings.gender_determination_locus_y,
+        help='Coordinates for includes region on chromosome X (default = gender_determination_locus_y in settings.py)'
+    )
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/check_gender_bam.py:43]: Let's leave that to the next feature in which I need to address gender determination anyways (Azure ticket 147348).

File: ExomeDepth/ed_csv_to_vcf.py [hunk 19]
```diff
@@ -128,48 +136,56 @@ def cnv_locationtype(region, par1, par2):
...
-                print("WARNING: {sample} chromosome Y CNV detected (region = {region}) in female, calc_copynumber set to 0 (deletion call) or 1 (duplication call)".format(
-                       sample = str(args.sampleid),
-                       region = str("_".join(str(x) for x in region))
-                      ))    
+                print(
+                    "WARNING: {sample} chromosome Y CNV detected (region = {region}) in female,"
+                    " calc_copynumber set to 0 (deletion call) or 1 (duplication call)".format(
+                       sample=str(args.sampleid),
+                       region=str("_".join(str(x) for x in region))
+                      ))
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/ed_csv_to_vcf.py:150]: Applied for all 3 warnings

File: ExomeDepth/exomedepth_db.py [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+    return entry
+
+
+def add_sample_to_db(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(args.sample_id, flowcell_id, args.refset)
+            store_sample(Session, entry)
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/exomedepth_db.py:31]: We should blame the original coder for this horrible code... ow wait, that's you! ;)
Changed both here and in other scripts

File: ExomeDepth/exomedepth_db.py [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+        refset = (
+            session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+        print(refset)
+        return refset
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/exomedepth_db.py:119]: No, if sample is already in database, it will set refset to the one in the database, and will not use argument.
If sample is not present in database, then it will use argument refset.

Will clarify the naming

[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/exomedepth_db.py:119]: add_sample_to_db_and_return_refset is a bit of legacy function as add_sample_to_db_and_return_refset_bam is used in DxNextflowWES and igv_xml_session.py.

Removed add_sample_to_db_and_return_refset completely

File: ExomeDepth/exomedepth_db.py (hunk 1 above)
Comment (melferink) [reply ExomeDepth/exomedepth_db.py:135]: add_sample_to_db_and_return_refset has been removed

File: ExomeDepth/igv_xml_session.py (hunk 4 above)
Comment (melferink) [reply ExomeDepth/igv_xml_session.py:12]: In principle not, as last column is not parsed. But will be better, so inlcuded.

File: ExomeDepth/igv_xml_session.py (hunk 6 above)
Comment (melferink) [reply ExomeDepth/igv_xml_session.py:45]: yes. Together with statistic this will be the column name in the igv track (ratio_test or log2ratio_test)

File: ExomeDepth/igv_xml_session.py (hunk 8 above)
Comment (melferink) [reply ExomeDepth/igv_xml_session.py:86]: Done, also for make_single_igv_session

File: ExomeDepth/submit_batch_exomedepth.py (hunk 16 above)
Comment (melferink) [reply ExomeDepth/submit_batch_exomedepth.py:220]: probably left over debug

File: ExomeDepth/submit_batch_exomedepth.py (hunk 17 above)
Comment (melferink) [reply ExomeDepth/submit_batch_exomedepth.py:244]: yes

File: ExomeDepth/run_ExomeDepth.py (hunk 14 above)
Comment (melferink) [reply ExomeDepth/run_ExomeDepth.py:264]: I don't see the complexity here. 
What do you suggest as a list of dictionaries?

Comment (rernst) [reply ExomeDepth/run_ExomeDepth.py:264]: It is complex because, it is long and order matters. It is used multiple times using different combinations of items in the list. I think that a dictionary would suite the usage if this list better.

Code would look like this:
```python
multiprocess_list.append({
    'model': model,
    'output_folder': output_folder,
   ...
})
```

**Review by rernst — CHANGES_REQUESTED** [review]
- Maybe take a look at ExonCov and the get_one_or_create -> https://github.com/UMCUGenetics/ExonCov/search?q=get_one_or_create
-  Move the print states in database.py to exomedepth_db.py? This makes reusing stuff easier.

File: ExomeDepth/database/functions.py [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+    sample_id = get_sample_id(bam)
+    flowcell_id = get_flowcelid_bam(bam)
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).all():
+            sample = Sample(sample=sample_id, flowcell=flowcell_id, refset=refset)
+            session.add(sample)
+            session.commit()
+        refset_db = (
+            session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/database/functions.py:117]: yes, refset should always be returned

File: ExomeDepth/submit_batch_exomedepth.py [hunk 23]
```diff
@@ -7,37 +7,30 @@
...
-from igv_xml_session import get_refset
+import database.functions
+import utils.utils
 import settings
 
 
-def determine_refset(args, bam, sampleid, refset_dic):
-    """ Determine refset to be used """
+def exomedepth_analysis(bam, args, gender_dic, suffix_dic, refset_dic):
+    bamfile = bam.rstrip("/").split("/")[-1]
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/submit_batch_exomedepth.py:16]: To remove a possible "/"  in the path. Split("/")[-1] would not be correct.

File: ExomeDepth/utils/utils.py [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+    for bam in bams:
+        skip_file_keywords = ['tmp', 'exomedepth', 'work']
+        if not any(keyword in bam.lower() for keyword in skip_file_keywords):  # Skip files in folder with keywords
+            with pysam.AlignmentFile(bam, "rb") as bam_file:
+                """ Extract sample_ID """
+                sample_id = re.split('_|\.', bam.split("/")[-1])[0]
+                """ Extract run ID """
+                run_id = bam.split("/")[-3]
+                if run_id == sample_id:  # Run is old IAP run
+                    run_id = bam.split("/")[-4]
```
[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/utils/utils.py:72]: No, current setup will work for 'wildcards', i.e. /hpc/diaggen/data/processed/exomes/2203 to process all samples from march 2022. If we use inputfolder that will not be possible (in the current setup).

File: ExomeDepth/submit_batch_exomedepth.py (hunk 23 above)
[the remark this answers is not in the document]
Comment (rernst) [reply ExomeDepth/submit_batch_exomedepth.py:16]: A file path will never end on `/`?

[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/submit_batch_exomedepth.py:16]: It does when you use tab-autocomplete....

[the remark this answers is not in the document]
Comment (melferink) [reply ExomeDepth/submit_batch_exomedepth.py:16]: Wait...this is a file not a path.. my bad

File: ExomeDepth/run_ExomeDepth.py [hunk 25]
```diff
@@ -19,60 +25,54 @@ def valid_read(read):
...
+            singularity_mnt=settings.singularity_mount_path,
+            singularity_container=settings.singularity_r_container,
+            refscript=settings.create_refset_r,
+            folder=mp_list[0],
+            outputid=mp_list[1],
+            targetbed=mp_list[2],
+            refgenome=settings.reference_genome,
+            exonbed=mp_list[3]
+            )
         )
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:67]: I think you update most to:
```python
action = (
    "export SINGULARITYENV_R_LIBS={r_libs} && singularity exec -B {singularity_mnt} "
    "{singularity_container} Rscript {refscript} {folder}/ {folder}/{outputid} {targetbed} {refgenome} {exonbed}\n"
).format(
    r_libs=settings.r_library_path,
    singularity_mnt=settings.singularity_mount_path,
    singularity_container=settings.singularity_r_container,
     refscript=settings.create_refset_r,
     folder=mp_list[0],
     outputid=mp_list[1],
     targetbed=mp_list[2],
     refgenome=settings.reference_genome,
     exonbed=mp_list[3]
)

```

File: ExomeDepth/run_ExomeDepth.py [hunk 26]
```diff
@@ -19,60 +25,54 @@ def valid_read(read):
...
 
+
 def make_refset(args):
 
     """Log all settings in setting.log file"""
-    log_file="{output}/settings.log".format(
-        output = args.output
+    log_file = "{output}/settings.log".format(
+        output=args.output
         )
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:76]: Out of scope, but you do not need multiple lines here.

File: ExomeDepth/settings.py.slangenspar [hunk 27]
```diff
@@ -58,25 +59,35 @@ locus_y = 'Y:2649520-59034050'
...
+    '22': 21, 'X': 22, 'Y': 23, 'MT': 24
+}
 
 # IGV session settings:
-template_xml = str(cwd)+"/igv_session_template.xml"
+template_single_xml = str(cwd)+"/templates/igv_session_template_single.xml"
+template_family_xml = str(cwd)+"/templates/igv_session_template_family.xml"
 igv_xml = str(cwd)+"/igv_xml_session.py"
-igv_settings = {"ratio":[0, 1, 2], "log2ratio":[-2.5, 0, 2.5] }
+igv_settings = {"ratio": [-0.1, 1, 2.1], "log2ratio": [-2.5, 0, 2.5]}
```
Comment (rernst) [inline ExomeDepth/settings.py.slangenspar:78]: Intended settings change?

File: ExomeDepth/submit_batch_exomedepth.py [hunk 28]
```diff
@@ -45,107 +57,192 @@ def process(bam, args, refset_dic, gender_dic, suffix_dic):
...
+    os.makedirs("{output}/logs".format(output=args.outputfolder), exist_ok=True)
+    os.makedirs("{output}/igv_tracks".format(output=args.outputfolder), exist_ok=True)
+    os.makedirs("{output}/UMCU/".format(output=args.outputfolder), exist_ok=True)
+    os.makedirs("{output}/HC/".format(output=args.outputfolder), exist_ok=True)
+
+    os.system("mv {output}/{sampleid}/*.log {output}/logs/".format(sampleid=sampleid, output=args.outputfolder))
+    os.system("mv {output}/{sampleid}/*.igv {output}/igv_tracks/".format(sampleid=sampleid, output=args.outputfolder))
+    os.system("mv {output}/{sampleid}/HC*.vcf {output}/HC/".format(sampleid=sampleid, output=args.outputfolder))
+    os.system("mv {output}/{sampleid}/UMCU*.vcf {output}/UMCU/".format(sampleid=sampleid, output=args.outputfolder))
+    os.system("rm -r {output}/{sample}".format(output=args.outputfolder, sample=sampleid))
```
Comment (rernst) [inline ExomeDepth/submit_batch_exomedepth.py:72]: There must by a `os` command for `rm` and `mv`.

**Review by rernst — CHANGES_REQUESTED** [review]
Hopefully the last small changes...

General remark:
- You do not need to str conversion when using `"_".join(....)`. There are more unnecessary type conversions. Guess it also a matter of code style, it does make it harder to read in my opinion.

File: ExomeDepth/settings.py.slangenspar (hunk 27 above)
Comment (melferink) [reply ExomeDepth/settings.py.slangenspar:78]: Yes. Solves a bug in IGV display

File: ExomeDepth/run_ExomeDepth.py [hunk 29]
```diff
@@ -36,14 +36,14 @@ def get_gender(bam):
...
+                "Sample {0} has a unknown gender based on chrX reads, but resolved as male based on sampleID"
+            ).format(bam_base)
             return "male"
         elif re.search('[C|P]F', os.path.basename(bam)):
-            print("Sample {0} has a unknown gender based on chrX reads, but resolved as female based on sampleID".format(
-                bam_base)
-            )
+            print(
+                "Sample {0} has a unknown gender based on chrX reads, but resolved as female based on sampleID"
+            ).format(bam_base)
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:46]: `.format()` links to `print()`?

```python
print(
    "Sample {0} has a unknown gender based on chrX reads, but resolved as female based on sampleID".format(bam_base)
)
```

Comment (rernst) [reply ExomeDepth/run_ExomeDepth.py:46]: Did you test this?

File: ExomeDepth/run_ExomeDepth.py [hunk 30]
```diff
@@ -147,65 +147,71 @@ def multiprocess_call(multiprocess_list):
...
+        r_igv_suffix=r_igv_suffix,
+        sampleid=args.sample
+    )
     os.system(action)
 
     """Perform csv to vcf conversion """
-    inputcsv = "{0}/{1}_{2}_{3}_{4}_exome_calls.csv".format(
+    inputcsv = (
+        "{0}/{1}_{2}_{3}_{4}_exome_calls.csv"
+    ).format(
```
Comment (rernst) [inline ExomeDepth/run_ExomeDepth.py:189]: Why the string between `()`? Only needed for multi line.

File: ExomeDepth/run_ExomeDepth.py (hunk 29 above)
Comment (melferink) [reply ExomeDepth/run_ExomeDepth.py:46]: Line-length made it newline, but forgot the additional ()

**Review by rernst — APPROVED** [review]
